src/Behat/Behat/Definition/Context/Attribute/DefinitionAttributeReader.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Definition\Context\Attribute;
+
+use Behat\Behat\Context\Attribute\AttributeReader;
+use Behat\Behat\Definition\Attribute\Definition;
+use Behat\Behat\Definition\Attribute\Given;
+use Behat\Behat\Definition\Attribute\Then;
+use Behat\Behat\Definition\Attribute\When;
+use ReflectionMethod;
+
+/**
+ * Reads definition Attributes from the context class.
+ *
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+class DefinitionAttributeReader implements AttributeReader
+{
+    /**
+     * @var string[]
+     */
+    private static $classes = array(
+        Given::class => 'Behat\Behat\Definition\Call\Given',
+        When::class  => 'Behat\Behat\Definition\Call\When',
+        Then::class  => 'Behat\Behat\Definition\Call\Then',
+    );
+
+    /**
+     * @{inheritdoc}
+     */
+    public function readCallees($contextClass, ReflectionMethod $method)
+    {
+        if (PHP_MAJOR_VERSION < 8) {
+            return [];
+        }
+
+        $attributes = $method->getAttributes(Definition::class, \ReflectionAttribute::IS_INSTANCEOF);
+
+        $callees = [];
+        foreach ($attributes as $attribute) {
+            $class = static::$classes[$attribute->getName()];
+            $callable = array($contextClass, $method->getName());
+
+            $callees[] = new $class($attribute->newInstance()->pattern, $callable, null);
```

Comment:
Description is `NULL` now by default, because we can't do docblock descriptions with Attributes.
We could opt to still check any annotations to get the description, but I don't really like that as Attributes are supposed to replace docblocks, not to live side-by-side.

Would be interested to hear what others think.

Reply:
From what I can gather they only seem to be used when running `[vendor/]bin/behat --definitions l`

Reply:
well, the description is still perfectly stored in the docblock, as it is about showing the documentation of the method

Reply:
True. I'll see if I can grab the docblock from the `ReflectionMethod` here and fetch the description from it.

Reply:
Done. I've introduced a `DocBlockHelper` to avoid having the exact same logic in two spots.